cmd/xamarin.go

```diff
@@ -72,121 +105,41 @@ and then hit Enter`
 	if err != nil {
 		return printXamarinScanFinishedWithError("Failed to analyze Xamarin solution: %s", err)
 	}
-	log.Debugf("xamSln: %#v", xamSln)
-
-	// filter only the iOS "app"" projects
-	xamarinProjectsToChooseFrom := []project.Model{}
-	for _, aXamarinProject := range xamSln.ProjectMap {
-		switch aXamarinProject.SDK {
-		case constants.SDKIOS, constants.SDKTvOS, constants.SDKMacOS:
-			if aXamarinProject.OutputType == "exe" {
-				// possible project
-				xamarinProjectsToChooseFrom = append(xamarinProjectsToChooseFrom, aXamarinProject)
-			}
-		default:
-			continue
-		}
-	}
-	log.Debugf("%d xamarinProjectsToChooseFrom: %#v", len(xamarinProjectsToChooseFrom), xamarinProjectsToChooseFrom)
 
-	// Xamarin Project
-	selectedXamarinProject := project.Model{}
-	{
-		if len(xamarinProjectsToChooseFrom) < 1 {
-			return printXamarinScanFinishedWithError(
-				"No acceptable Project found in the provided Solution, or none can be used for iOS Archive.",
-			)
+	if enableVerboseLog {
+		b, err := json.MarshalIndent(xamSln, "", "\t")
+		if err == nil {
+			log.Debugf("xamarin solution:\n%s", b)
 		}
+	}
 
-		if paramXamarinProjectName != "" {
-			// project specified via flag/param
-			for _, aProj := range xamarinProjectsToChooseFrom {
-				if paramXamarinProjectName == aProj.Name {
-					selectedXamarinProject = aProj
-					break
-				}
-			}
-			if selectedXamarinProject.Name == "" {
-				return printXamarinScanFinishedWithError(
-					`Invalid Project specified (%s), either not found in the provided Solution or it can't be used for iOS "Archive for Publishing".`,
-					paramXamarinProjectName)
-			}
-		} else {
-			// no project CLI param specified
-			if len(xamarinProjectsToChooseFrom) == 1 {
-				selectedXamarinProject = xamarinProjectsToChooseFrom[0]
-			} else {
-				projectNames := []string{}
-				for _, aProj := range xamarinProjectsToChooseFrom {
-					projectNames = append(projectNames, aProj.Name)
-				}
-				sort.Strings(projectNames)
+	archivableSolutionConfigNames := archivableSolutionConfigNames(xamSln.ProjectMap)
 
-				fmt.Println()
-				answerValue, err := goinp.SelectFromStringsWithDefault(`Select the Project Name you use for "Archive for Publishing" (usually ends with ".iOS", e.g.: MyProject.iOS)?`, 1, projectNames)
-				if err != nil {
-					return printXamarinScanFinishedWithError("Failed to select Project: %s", err)
-				}
-				log.Debugf("selected project: %v", answerValue)
-				for _, aProj := range xamarinProjectsToChooseFrom {
-					if answerValue == aProj.Name {
-						selectedXamarinProject = aProj
-						break
-					}
-				}
-			}
-		}
+	if len(archivableSolutionConfigNames) < 1 {
+		return printXamarinScanFinishedWithError(`No acceptable Configuration found in the provided Solution and Project, or none can be used for iOS "Archive for Publishing".`)
```

Comment:
Implement builtin `error` interface instead of calling a function which is printing and returning an error.
```golang
type ArchiveError struct {
  toolName string
  msg string
}

func (e ArchiveError) Error() string
```